test/junit/scala/collection/IteratorTest.scala

```diff
@@ -484,4 +484,9 @@ class IteratorTest {
     assertSameElements(List(10,11,13), scan)
     assertSameElements(List(10,-1,-1,-11,11,-2,-2,-13,13,-3), results)
   }
+
+  @Test def unfold(): Unit = {
+    val it = Iterator.unfold(1)(i => if (i > 10) None else Some((i, i + 1)))
+    assertSameElements(1 to 10, it)
```

Comment:
Do you mind adding a test that produces an empty `Iterator`?

Reply:
What about a test checking that creation of Iterator.UnfoldIterator and invoking View.Unfold.iterator doesn't invoke generator function (i.e. one that takes state and produces optionally next element and state)? LazyList.unfold doesn't have this property - it always invokes generator function once, even if the LazyList is never used after LazyList.unfold.

Reply:
@tarsa that's a good idea - can also do that :)